### DISCUSSION TITLE
Issue with canvas

### DISCUSSION THREAD

**Discussion by sajrashid** [gatsbyjs/gatsby#32167] [Help]
## Description

Same as this issue https://github.com/gatsbyjs/gatsby/issues/17661 with canvas

Describe the issue that you're seeing.

```
6:25:38 AM: success Building production JavaScript and CSS bundles - 37.833s
6:25:38 AM: success Rewriting compilation hashes - 0.012s
6:25:45 AM: error Generating SSR bundle failed
6:25:45 AM: Unexpected character '' (1:0)
6:25:45 AM: not finished Generating image thumbnails - 55.522s
6:25:45 AM: not finished Building static HTML for pages - 7.013s
6:25:45 AM: error Command failed with exit code 1.
```

tried the suggested changes in gatsby-node.js, the issue happens when creating a production release 
### Steps to reproduce

Clear steps describing how to reproduce the issue.
try   "build": "gatsby build" from this repo
https://github.com/sajrashid/docsSite
you can also see the issue [here](https://app.netlify.com/teams/sajrashid/builds/60dbffaea83e8a0008acbb8d) on netlify 

How to Make a Minimal Reproduction: https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/

Run `gatsby info --clipboard` in your project directory and paste the output
 ```
 Binaries:
    Node: 16.4.0 - C:\Program Files\nodejs\node.EXE    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD 
    npm: 7.17.0 - C:\Program Files\nodejs\npm.CMD  
  Languages:
    Python: 3.9.5 - /c/Python39/python
  Browsers:
    Edge: Spartan (44.19041.1023.0), Chromium      
(91.0.864.59)
  npmPackages:
    gatsby: ^2.26.1 => 2.26.1
    gatsby-image: ^2.10.0 => 2.10.0
    gatsby-plugin-sharp: ^2.13.4 => 2.13.4
    gatsby-remark-embed-snippet: ^6.5.0 => 6.5.0   
    gatsby-remark-embedder: ^5.0.0 => 5.0.0        
    gatsby-remark-highlight-code: ^3.0.0 => 3.0.0  
    gatsby-remark-prismjs: ^5.5.0 => 5.5.0
    gatsby-source-filesystem: ^2.10.0 => 2.10.0    
    gatsby-transformer-remark: ^2.15.0 => 2.15.0   
    gatsby-transformer-sharp: ^2.11.0 => 2.11.0    
  npmGlobalPackages:
    gatsby-cli: 3.8.0
```

Also name any `flags` you use inside `gatsby-config.js`.
not sure can't see anything called flags

**Comment by LekoArts** [comment]
Your `onCreateWebpackConfig` call is inside the `createPages` API call. https://github.com/sajrashid/docsSite/blob/0d611f67f6b2ded02c53b5dfc38deca863fbccb1/gatsby-node.js#L25-L56

You need to place this at the top level.

**Reply by sajrashid** [reply]
oops very sorry, my bad 

I've moved it out and deployed it fails with the same :

direct link to my [gastby-node.js](https://github.com/sajrashid/docsSite/blob/master/gatsby-node.js)  file

8:36:58 AM: error Generating SSR bundle failed
8:36:58 AM: Unexpected character '' (1:0)

[build log](https://app.netlify.com/teams/sajrashid/builds/60dc1e99ca95080008ca1933)

**Comment by sajrashid** [comment]
the example I pasted contained "develop-html"  changed to  "build-html" and it's working fine.. thanks

**Comment by sajrashid** [comment]
ensure `if (stage === "build-html") {   ` & not `"develop-html"`
